services/batch-service.js

```diff
@@ -158,18 +157,15 @@ class BatchService {
     }
 
     async deleteBatchByFilter(filter) {
-        return await this.batchCollection.deleteMany(filter);
+        return await this.batchDAO.deleteByFilter(filter);
     }
 
     async findByID(id) {
         return await this.batchDAO.findById(id);
     }
     // private function
     async _getBatchDisplayID(submissionID) {
-        const pipeline = [{$match: {submissionID}}, {$count: "total"}];
-        const batches = await this.batchCollection.aggregate(pipeline);
-        const totalDocs = batches.pop();
-        return totalDocs?.total + 1 || 1;
+        return await this.batchDAO.getNextDisplayID(submissionID);
```

Comment:
This looks like it returns 1 by default. Could you double check?
return totalDocs?.total + 1 || **1**;